src/Emulator/Peripherals/Peripherals/Mocks/HPSHostController.cs

```diff
@@ -334,27 +347,154 @@ private bool RXNECleared()
             return !currentSlave.RxNotEmpty;
         }
         
-        private void PollForRegisterBit(RegisterBitName bitName)
-        {
-            int i = 0;
+        private void PollForRegisterBit(RegisterBitName bitName){
             Func<bool> registerAccess;
-            switch(bitName)
-            {
+            int i = 0;
+            switch(bitName){
                 case RegisterBitName.OA1EN:
                     registerAccess = OA1ENEnabled;
                     break;
                 case RegisterBitName.RXNE:
                     registerAccess = RXNECleared;
                     break;
                 default:
-                    throw new RecoverableException("Register bit name does not exist.");
+                    registerAccess = null;
+                    break;
             }
+            while(!registerAccess() && i < 8){
+                Thread.Sleep(500);
+                i ++;
+            }
+        }
+
+        private void PollForError(TimeInterval timeInterval){
```

Comment:
Since this method does not return any status how do you know if reading was successful or not?

Reply:
The reason for this method is to read the Error register multiple times in case the error register wasn't updated in time _if_ there's any error that'll be updated in error register, but not necessarily always. Hence here it's just a void method to read the error register multiple times, if after several times of read it is still 0, then we can be relatively confident to say that the error register to be read contains only 0s.